aiohttp/cookiejar.py

```diff
@@ -261,8 +261,43 @@ def filter_cookies(self, request_url: URL = URL()) -> "BaseCookie[str]":
                 request_origin = request_url.origin()
             is_not_secure = request_origin not in self._treat_as_secure_origin
 
+        # d: domain, d could be subdomain
+        # p: path
+        d = hostname
+
+        pairs = set()
+        while d:
+            p = request_url.path
+            while p:
+                pairs.add((d, p))
+                path_leftovers = p.rsplit("/", maxsplit=1)
+                # handle last element for rsplit
+                if len(path_leftovers) > 1:
+                    p = path_leftovers[0]
+                else:
+                    p = None
+
+            # handle last element for split
+            leftovers = d.split(".", maxsplit=1)
+            if len(leftovers) > 1:
+                d = leftovers[-1]
+            else:
+                d = None
```

Comment:
For this one, maybe would be more efficient as:
```suggestion
            try:
                d = d.split(".", maxsplit=1)[1]
            except IndexError:
                d = ""
```